docs/architecture/decisions/0008-file-upload.md

```diff
@@ -0,0 +1,54 @@
+# 8. File upload
+
+Date: 2025-09-30
+
+## Status
+
+Accepted
+
+## Context
+
+Our registered users will need to upload a Summary Log that is in Excel. These files come from external users, so there is a possibility that they are unsasfe. We need to:
+
+- accepted the file upload
+- scan for viruses before we store them
+- stored in a reliable location for later validation/processing
+
+The Core Delivery Platform (CDP) provides a managed Uploader service that does exactly this. It takes care of file acceptance, virus scanning and storage in S3. Using it means we don’t have to build our own file handling pipeline.
+
+## Decision
+
+We will integrate with CDP Uploader for our file uploads.
+
+The flow works like this:
+
+- The frontend asks CDP to start an upload, and CDP gives back an `uploadId` and URL.
+- The operator uploads their Excel file directly to CDP.
+- CDP runs an async virus scan.
+- CDP stores the file in our S3 bucket.
+- If the file is clean, CDP calls back to the frontend.
```

Comment:
Is there a notification if it fails the scan?

Reply:
@GraemeF the user should be on a page that is polling the server, if the file is rejected they should see a message on the page returned by the server. If the user leaves that page then we do have mechanisms to notify the user either via GovNotify or on screen the next time they log in. That said, it might be better to shift the discussion to the LLD digram that this was summarised from, as luck would have it - [the PR is still open](https://github.com/DEFRA/epr-backend/pull/119)

Reply:
Ah, I see on the diagram we get `200: { status: 'rejected' }` when we check the status 👍